examples/subscribe.rs

```diff
@@ -26,60 +33,122 @@ async fn main() -> Result<(), Box<dyn snafu::Error>> {
             secret_key: None,
         })
         .with_user_id("user_id")
+        .with_filter_expression("some_filter")
+        .with_heartbeat_value(100)
+        .with_heartbeat_interval(5)
         .build()?;
 
     println!("running!");
 
-    let subscription = client
-        .subscribe()
+    client
+        .set_presence_state(HashMap::<String, String>::from([
+            (
+                "is_doing".to_string(),
+                "Nothing... Just hanging around...".to_string(),
+            ),
+            ("flag".to_string(), "false".to_string()),
+        ]))
         .channels(["my_channel".into(), "other_channel".into()].to_vec())
-        .heartbeat(10)
-        .filter_expression("some_filter")
-        .execute()?;
+        .user_id("user_id")
+        .execute()
+        .await?;
+
+    tokio::time::sleep(tokio::time::Duration::from_secs(3)).await;
+
+    let subscription = client.subscription(
+        Some(&["my_channel", "other_channel"]),
+        None,
+        Some(vec![SubscriptionOptions::ReceivePresenceEvents]),
+    );
```

Comment:
It seems to be a little bit unreadable. Can you provide at least the `SubscriptionParams` or something like that?

Reply:
But method itself provides documentation to the arguments? It is not enough? `SubscriptionParams` used by `EventEngine` already, and they are different from the set of arguments for this method.

Reply:
It's more about convention. 
I prefer when you exactly know what do you try to achieve. 
That's why I propose `SubscriptionParams` or just a builder for that. 

It will explicit explain what options you set here.

Reply:
Added `SubscriptionParams`